lib/common/patchset_display.c

```diff
@@ -301,19 +348,27 @@ pcmk__xml_log_patchset(uint8_t log_level, const xmlNode *patchset)
     crm_element_value_int(patchset, "format", &format);
     switch (format) {
         case 1:
-            xml_show_patchset_v1(out, patchset);
+            if (log_level < LOG_DEBUG) {
+                rc = xml_show_patchset_v1(out, patchset,
+                                          pcmk__xml_fmt_pretty
+                                          |pcmk__xml_fmt_diff_short);
+            } else {    // Note: LOG_STDOUT > LOG_DEBUG
+                rc = xml_show_patchset_v1(out, patchset, pcmk__xml_fmt_pretty);
+            }
             break;
         case 2:
-            xml_show_patchset_v2(out, patchset);
+            rc = xml_show_patchset_v2(out, patchset);
             break;
         default:
             crm_err("Unknown patch format: %d", format);
+            rc = pcmk_rc_unknown_format;
```

Comment:
By the way, this is a misuse of the return code the way it's currently mapped to a string in `results.c`: `"Unknown output format"`. It seems reasonable to use it more broadly, so maybe we could change that to simply say `"Unknown format"`. It's already being used in ways other than unknown output format throughout libcrmcommon.

Reply:
I'd rather keep unknown_format for its original purpose (--output-as set to something unrecognized), and add a new code or codes for all the current uses that don't fit that (looks like we were just being lazy). However it's not important enough to worry about at this point, I'll just create a task anyone can pick up later.